flow-typed/npm/redux-loop_v4.x.x.js

```diff
@@ -0,0 +1,176 @@
+    reducer: LoopReducer<S, A>,
+    preloadedState: S,
+    enhancer: StoreEnhancer<S>
+  ) => Store<S>;
+
+  declare export type StoreCreatorExplicit<S, A> = (
+    reducer: LoopReducer<S, A>,
+    preloadedState: S,
+    enhancer: StoreEnhancer<S>
+  ) => Store<S>;
+
+  declare export type Loop<S, A> = [S, CmdType<A>];
+
+  declare export type LoopReducer<S, A> = (
+    state: S | void,
+    action: A,
+    ...args: any[]
+  ) => Loop<S, A> | S;
+
+  declare export interface LiftedLoopReducer<S, A> {
+    (state: S | void, action: A, ...args: any[]): Loop<S, A>;
+  }
+
+  declare export type CmdSimulation<R> = {
+    result: R,
+    success: boolean
+  };
+
+  declare export interface MultiCmdSimulation {
+    [index: number]: CmdSimulation<any> | MultiCmdSimulation;
+  }
+
+  declare export interface NoneCmd {
+    type: "NONE";
+    simulate(): null;
+  }
+
+  declare export interface ListCmd<A> {
+    type: "LIST";
+    cmds: CmdType<A>[];
+    sequence?: boolean;
+    batch?: boolean;
+    simulate(simulations: MultiCmdSimulation): A[];
+  }
+
+  declare export interface ActionCmd<A> {
+    type: "ACTION";
+    actionToDispatch: A;
+    simulate(): A;
+  }
+
+  declare export interface MapCmd<A, B = any, C = any> {
+    type: "MAP";
+    tagger: ((...args:Array<C|B>) => A) | ((subAction: B) => A);
+    nestedCmd: CmdType<A>;
+    args: Array<C>;
+    simulate(simulations?: CmdSimulation<any> | MultiCmdSimulation): A[] | A | null;
+  }
+
+  declare export interface RunCmd<A, B, C> {
+    type: "RUN";
+    func: (...args: Array<C>) => B;
+    args?: Array<C>;
+    failActionCreator?: ActionCreator<A, any>;
+    successActionCreator?: ActionCreator<A, B>;
+    forceSync?: boolean;
+    simulate(simulation: CmdSimulation<B>): A;
+  }
+
+  declare export type SequenceCmd<A> = ListCmd<A>;
+  declare export type BatchCmd<A> = ListCmd<A>;
+
+  declare export type CmdType<A, B = any, C = any> =
+    | ActionCmd<A>
+    | ListCmd<A>
+    | MapCmd<A>
+    | NoneCmd
+    | RunCmd<A, B, C>
+    | BatchCmd<A>
+    | SequenceCmd<A>;
+
+  declare export function install<S, A>(): StoreEnhancer<S, A>;
+
+  declare export function loop<S, A>(state: S, cmd: CmdType<A>): Loop<S, A>;
+
+  declare function Cmd$action<A>(action: A): ActionCmd<A>;
+  declare function Cmd$batch<A>(cmds: CmdType<A>[]): BatchCmd<A>;
+  declare function Cmd$sequence<A>(cmds: CmdType<A>[]): SequenceCmd<A>;
+  declare function Cmd$list<A>(
+    cmds: CmdType<A>[],
+    options?: {
+      batch?: boolean,
+      sequence?: boolean,
+      testInvariants?: boolean
+    }
+  ): ListCmd<A>;
+
+  declare function Cmd$map<A, B>(
+    cmd: CmdType<B>,
+    tagger: (subAction: B) => A
+  ): MapCmd<A>;
+  declare function Cmd$map<A, B, C>(
+    cmd: CmdType<B>,
+    tagger: (arg1: C, subAction: B) => A,
+    args?: Array<C>
+  ): MapCmd<A>;
+  declare function Cmd$map<A, B, C>(
+    cmd: CmdType<B>,
+    tagger: (arg1: C, arg2: C, subAction: B) => A,
```

Comment:
if it would be in the end than yes - I could use rest

```ts
tagger: (subAction: B, ...rest: C[]) => A,
```

How much maximum arguments we want? Up to 9?

Reply:
i don't see a reason to impose a limit. a rest parameter would allow any number

Reply:
Oh I see. because the rest parameter would need to be first? In that case, let's leave it at 3 and worry about adding more later.